ecl/hql/hqlattr.cpp

```diff
@@ -2813,6 +2814,27 @@ IHqlExpression * calcRowInformation(IHqlExpression * expr)
                 info.limitMin(choosenLimit);
         }
         break;
+    case no_quantile:
+        {
+            __int64 parts = getIntValue(expr->queryChild(1), 0);
+            if ((parts > 0) && !isGrouped(expr) && !isLocalActivity(expr))
+            {
+                if (expr->hasAttribute(firstAtom))
+                    parts++;
+                if (expr->hasAttribute(lastAtom))
+                    parts++;
+
+                IHqlExpression * transform = queryNewColumnProvider(expr);
+                if (transformContainsSkip(transform) || expr->hasAttribute(dedupAtom))
+                    info.setRange(0,parts-1);
+                else
+                    info.setN(parts-1);
+            }
+            else
+                info.setUnknown(RCMfew);
```

Comment:
If it's grouped, isn't the expected rowcount some fraction of the input's rowcount (i.e. not necessarily few) ?